provider/azure/internal/ad/client.go

```diff
@@ -69,8 +77,30 @@ func WithOdataErrorUnlessStatusCode(codes ...int) autorest.RespondDecorator {
 
 				// Copy and replace the Body in case it does not contain an error object.
 				// This will leave the Body available to the caller.
-				b, decodeErr := autorest.CopyAndDecode(autorest.EncodedAsJSON, resp.Body, &oe)
+				var b bytes.Buffer
+				if _, err := io.Copy(&b, resp.Body); err != nil {
+					return err
+				}
+				var decodeErr error
```

Comment:
> BTW I think it might be better to strip the BOM in all JSON decoding code - surely
this isn't the only place that can include it?

Maybe, but we don't control the other code. That will need to go in go-autorest. The reason this code exists at all is because the AAD (Azure Active Directory) API endpoint is a little different to the other Azure API endpoints. Different error structures. So I wouldn't be surprised if this *is* the only place that the BOM shows up, because it's probably getting routed to a different backend.